app/livechat/server/business-hour/BusinessHourManager.ts

```diff
@@ -17,100 +17,85 @@ const cronJobDayDict: Record<string, number> = {
 };
 
 export class BusinessHourManager {
-	private businessHour: IBusinessHour;
+	private types: Map<string, IBusinessHourType> = new Map();
+
+	private behavior: IBusinessHourBehavior;
 
 	private cronJobs: ICronJobs;
 
 	private cronJobsCache: string[] = [];
 
-	constructor() {
+	constructor(cronJobs: ICronJobs) {
+		this.cronJobs = cronJobs;
+		callbacks.run('onCreateBusinessHourManager', this);
 		this.openWorkHoursCallback = this.openWorkHoursCallback.bind(this);
 		this.closeWorkHoursCallback = this.closeWorkHoursCallback.bind(this);
 	}
 
-	onStartBusinessHourManager(businessHour: IBusinessHour, cronJobs: ICronJobs): void {
-		this.cronJobs = cronJobs;
-		this.registerBusinessHourMethod(businessHour);
-	}
-
-	registerBusinessHourMethod(businessHour: IBusinessHour): void {
-		this.businessHour = businessHour;
-	}
-
-	async dispatchOnStartTasks(): Promise<void> {
+	async onStartManager(): Promise<void> {
 		await this.createCronJobsForWorkHours();
-		await this.openBusinessHoursIfNeeded();
-	}
-
-	async dispatchOnCloseTasks(): Promise<void> {
-		await this.removeBusinessHoursFromAgents();
-		await this.removeCronJobs();
+		this.setupCallbacks();
+		this.behavior.onStartBusinessHours();
 	}
 
-	async saveBusinessHour(businessHourData: ILivechatBusinessHour): Promise<void> {
-		await this.businessHour.saveBusinessHour(businessHourData);
-		if (!settings.get('Livechat_enable_business_hours')) {
-			return;
-		}
-		await this.dispatchOnStartTasks();
-	}
-
-	async getBusinessHour(id?: string): Promise<ILivechatBusinessHour | undefined> {
-		return this.businessHour.getBusinessHour(id as string);
+	async onCloseManager(): Promise<void> {
```

Comment:
Wouldn't it be nice to remove the callbacks when the server is closed?
Also, I'd call this just method as `stopManager`.